sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -93,90 +93,56 @@ def generate_pip_requirements(self, target_file):
+    if python_version is 'python3':
+      f.write('RUN apt-get update -y && apt-get install --no-install-recommends -y -q python3 python3-pip python3-setuptools\n')
+    else:
+      f.write('RUN apt-get update -y && apt-get install --no-install-recommends -y -q python python-pip python-setuptools\n')
+    if requirement_filename is not None:
+      f.write('ADD ' + requirement_filename + ' /ml/requirements.txt\n')
       if python_version is 'python3':
-        f.write('ENTRYPOINT ["python3", "/ml/' + python_filepath + '"]')
+        f.write('RUN pip3 install -r /ml/' + requirement_filename + '\n')
       else:
-        f.write('ENTRYPOINT ["python", "/ml/' + python_filepath + '"]')
-
-  def _wrap_files_in_tarball(self, tarball_path, files={}):
-    """ _wrap_files_in_tarball creates a tarball for all the input files
-    with the filename configured as the key of files """
-    if not tarball_path.endswith('.tar.gz'):
-      raise ValueError('the tarball path should end with .tar.gz')
-    with tarfile.open(tarball_path, 'w:gz') as tarball:
-      for key, value in files.items():
-        tarball.add(value, arcname=key)
-
-  def prepare_docker_tarball_with_py(self, arc_python_filename, python_filepath, base_image, local_tarball_path, python_version, dependency=None):
-    """ prepare_docker_tarball is the API to generate dockerfile and prepare the tarball with python scripts
-    args:
-      python_version (str): choose python2 or python3
-    """
-    if python_version not in ['python2', 'python3']:
-      raise ValueError('python_version has to be either python2 or python3')
-    with tempfile.TemporaryDirectory() as local_build_dir:
-      has_requirement_file = False
-      local_requirement_path = os.path.join(local_build_dir, self._ARC_REQUIREMENT_FILE)
-      if dependency is not None and len(dependency) != 0:
-        self._generate_pip_requirement(dependency, local_requirement_path)
-        has_requirement_file = True
-      local_dockerfile_path = os.path.join(local_build_dir, self._arc_dockerfile_name)
-      self._generate_dockerfile_with_py(local_dockerfile_path, base_image, arc_python_filename, has_requirement_file, python_version)
-      file_lists =  {self._arc_dockerfile_name:local_dockerfile_path,
-                     arc_python_filename:python_filepath}
-      if has_requirement_file:
-        file_lists[self._ARC_REQUIREMENT_FILE] = local_requirement_path
-      self._wrap_files_in_tarball(local_tarball_path, file_lists)
-
-  def prepare_docker_tarball(self, dockerfile_path, local_tarball_path):
-    """ prepare_docker_tarball is the API to prepare a tarball with the dockerfile """
-    self._wrap_files_in_tarball(local_tarball_path, {self._arc_dockerfile_name:dockerfile_path})
+        f.write('RUN pip install -r /ml/requirements.txt\n')
+    f.write('ADD ' + entrypoint_filename + ' /ml/main.py\n')
+    if python_version is 'python3':
+      f.write('ENTRYPOINT ["python3", "/ml/main.py"]')
```

Comment:
you probably need to add -u to force the logs to be flushed out to stdout ASAP.

Reply:
Good idea.